File: src/main/java/BTMaker/BTMaker/ResourceType.java
package BTMaker.BTMaker;

public enum ResourceType {
    IMAGE(2),
    MIDI(3),
    STRINGS(4),
    LAYOUT(5),
    LEVEL(8);

    public final byte code;

    ResourceType(int code) {
        this.code = (byte) code;
    }

    public static ResourceType fromCode(byte code) {
        for (ResourceType type : values()) {
            if (type.code == code) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown ResourceType code: " + code);
    }
}

File: src/main/java/BTMaker/BTMaker/resources/ResourceType.kt
package BTMaker.BTMaker.resources

enum class ResourceType {
    IMAGE,
    MIDI,
    STRINGS,
    LAYOUT,
    LEVEL;

    companion object {
        fun fromCode(code: Number): ResourceType {
            return when (code) {
                2 -> IMAGE
                3 -> MIDI
                4 -> STRINGS
                5 -> LAYOUT
                8 -> LEVEL
                else -> throw IllegalArgumentException("Unknown resource type code: $code")
            }
        }
    }
}